src/linker/pipeline.py

```diff
@@ -95,6 +96,7 @@ def _validate(self) -> None:
         for validation in [
             self._validate_pipeline,
             self._validate_implementations,
+            self._validate_input_data,
```

Comment:
Isn't this cleaner?

```python
validates = True
for validation in [...]:
    validations &= validation()

if not validates:
    ...
```

Reply:
Or maybe better is this:

```python
def _validate_X(*args) -> Dict[str, str]:
    """ 
    validates X and returns a dict of all validation error messages keyed by X.
    An empty dict means there are no errors. """
    ...

errors = {
    **validator() for validator in [...]
}
if errors:
    ...
```

Reply:
Maybe this is out of scope for this PR, but if you agree and it's quick enough you could make the change.

Reply:
Pylance tells me that you can't do a dict expansion within a comprehension like that. WDYT about the change I just made?

Reply:
That is super weird to me, but yeah it seems you can't do that.

Reply:
What about this instead?
```python
errors = {
    **self._validate_pipeline(),
    **self._validate_implementation(),
    **self._validate_input_data(),
}
```

Reply:
sure lgtm, done